subprojects/file-collections/src/main/java/org/gradle/api/internal/file/collections/GeneratedSingletonFileTree.java

```diff
@@ -101,10 +100,7 @@ public DirectoryFileTree getMirror() {
 
     @Override
     public void visitStructure(MinimalFileTreeStructureVisitor visitor, FileTreeInternal owner) {
-        if (visitor.prepareForVisit(this) != FileCollectionStructureVisitor.VisitType.NoContents) {
-            // TODO: Fail when using NoContents when we moved to using the new file system watching infrastructure for continuous build.
```

Comment:
should we fail here as the comment suggests?

Reply:
Removing is better, so you can't do it wrong.

Reply:
Prepare was removed.